.github/workflows/azure_function.yml

```diff
@@ -0,0 +1,31 @@
+name: Azure function
+
+on: workflow_call
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install and configure Poetry
+        uses: snok/install-poetry@v1
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.x"
+      - name: Install dependencies
+        run: |
+          poetry install --all-extras
+      - name: Update function requirements
+        run: poetry export --without-hashes --format=requirements.txt > examples/function/requirements.txt
+      - name: Install dependencies
+        working-directory: ./examples/function
+        run: |
+          poetry install --all-extras
+          poetry run build
+      - name: upload_artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: acme_dns_azure_function
+          path: ./examples/function/dist
```

Comment:
Ensure plugins are installed and export requirements.txt on build
Only publish zip archive
```
      - name: Install dependencies
        working-directory: ./examples/function
        run: |
          poetry install --all-extras
          poetry run add-plugins
          poetry run build
      - name: upload_artifact
        uses: actions/upload-artifact@v4
        with:
          name: acme_dns_azure_function
          path: ./examples/function/dist/*.zip
```